app/services/CheckoutService.scala

```diff
@@ -328,22 +330,53 @@ class CheckoutService(identityService: IdentityService,
         autoRenew = renewal.plan.charges.billingPeriod.isRecurring,
         fastForwardTermStartDate = currentVersionExpired // If the sub has expired, we need to shift the term dates forward
       )
+      val finalRenewCommand = applyPromoIfNecessary(contact, basicRenewCommand)
+      logger.info(s"Final renew command will be: ${finalRenewCommand}")
+      finalRenewCommand
```

Comment:
but you need to import the withLogging command from LogImplicits._ in membership common